src/pages/SignupHandler/index.js

```diff
@@ -15,236 +15,269 @@ import { routerRedux } from 'dva/router';
 import { connect } from 'dva';
 import styles from './index.less';
 
-@connect(auth => ({
-  auth: auth.auth,
+@connect(({ datastore }) => ({
+  datastoreConfig: datastore.datastoreConfig,
 }))
 class SignupHandler extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
+      firstName: '',
+      lastName: '',
       username: '',
       password: '',
-      confPassword: '',
-      email: '',
-      variantVal: 'tertiary',
-      btnColor: 'black',
-      btnDisabled: 'true',
-      charLength: ['block', 'none'],
-      specialChar: ['block', 'none'],
-      number: ['block', 'none'],
-      upperCaseChar: ['block', 'none'],
-      passwordMatch: '',
-      emailValidation: '',
+      // eslint-disable-next-line react/no-unused-state
+      confirmedPassword: '',
+      submitButtonDisabled: true,
+      submitButtonColor: 'black',
+      passwordValidation: {
+        charCount: false,
+        specialCharCount: false,
+        containsNumber: false,
+        containsUppercaseLetter: false,
+      },
+      passwordValidationMessage: '',
+      usernameValidationMessage: '',
     };
   }
 
-  componentDidMount = () => {
-    this.disableSubmitBtn();
-  };
+  componentDidMount() {
+    this.fetchDatastoreConfig();
+  }
 
-  enableSubmitBtn = () => {
-    this.setState({
-      variantVal: 'primary',
-      btnColor: 'white',
-      btnDisabled: 'false',
-    });
-  };
+  fetchDatastoreConfig = () => {
+    const { dispatch } = this.props;
 
-  disableSubmitBtn = () => {
-    this.setState({
-      variantVal: 'tertiary',
-      btnColor: 'black',
-      btnDisabled: 'true',
+    dispatch({
+      type: 'datastore/fetchDatastoreConfig',
     });
   };
 
-  handleNameInputChange = username => {
-    const { email, password, confPassword } = this.state;
-    this.setState({
-      username,
-    });
-    if (username !== '' && email !== '' && password !== '' && confPassword !== '') {
-      this.enableSubmitBtn();
+  registerUser = () => {
+    const { dispatch, datastoreConfig } = this.props;
+    const { firstName, lastName, username, password, confirmedPassword } = this.state;
+
+    let validRegistration = true;
+    const validEmailRegex = /\S+@\S+\.\S+/;
+
+    if (!validEmailRegex.test(username)) {
```

Comment:
Not a big deal in the context of this change, but isn't there a better email validation package available than just the character set check? (Even more irrelevant, I suppose, since Nikhil is using a strong email validator package for the server API.)